conans/client/tools/system_pm.py

```diff
@@ -138,16 +196,31 @@ def _get_package_names(self, packages, arch_names):
             arch = host_arch or build_arch
             parsed_packages = []
             for package in packages:
-                for package_name in package.split(" "):
-                    parsed_packages.append(self._tool.get_package_name(package_name, arch,
-                                                                       arch_names))
+                if isinstance(package, (tuple, list)):
+                    parsed_packages.append(tuple(self._get_package_names(package, arch_names)))
+                else:
+                    for package_name in package.split(" "):
+                        parsed_packages.append(self._tool.get_package_name(package_name, arch,
+                                                                           arch_names))
             return parsed_packages
         return packages
 
     def installed(self, package_name):
         return self._tool.installed(package_name)
 
+    def _to_be_installed(self, packages):
+        """ Returns a list with all not installed packages.
+        """
+        for pkg in packages:
+            if isinstance(pkg, (list, tuple)):
```

Comment:
According to the filter in the function `install_packages`, every `pkg` is a string, never a list. No need for this check.

Reply:
Done.